subprojects/common/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java

```diff
@@ -744,6 +751,21 @@ public void testArrayWrite() {
 		assertEquals(Int(100), Read(newArr, Int(182)).eval(ImmutableValuation.empty()));
 	}
 
+	@Test
```

Comment:
As far as I see, this test checks whether we can read(init(...)) can be simplified or not. Can you also include a test when the array init expression only contains literals and calling simplify makes it an arraylitexpr?

Reply:
I've added some LOC testing this behavior, but I also kept the existing tests - is it OK like this? https://github.com/ftsrg/theta/pull/128/commits/c81108dccfd8e3799c058554e36360f6e7fabfaa